trin-core/src/portalnet/overlay_service.rs

```diff
@@ -967,17 +997,46 @@ impl<TContentKey: OverlayContentKey + Send, TMetric: Metric + Send>
                 "[{:?}] Skipping processing for content connection ID {}",
                 self.protocol, id
             ),
-            Content::Content(_) => {
-                debug!(
-                    "[{:?}] Skipping processing for content bytes",
-                    self.protocol
-                )
-            }
+            Content::Content(content) => self.process_received_content(content, request).await,
             Content::Enrs(enrs) => self
                 .process_discovered_enrs(enrs.into_iter().map(|ssz_enr| ssz_enr.into()).collect()),
         }
     }
 
+    async fn process_received_content(&mut self, content: ByteList, request: FindContent) {
+        let content_key = match TContentKey::try_from(request.content_key) {
+            Ok(val) => val,
+            Err(msg) => {
+                error!("Unable to process received content: Invalid content key: {msg:?}");
```

Comment:
This is a weird situation. We requested content with an invalid content ID and then find out about it during processing the response, yeah?

The error message doesn't really convey how weird that is. I think a log like this might give me more up-front understanding:
```suggestion
                error!("Unable to process received content: we sent a request with an invalid content key: {msg:?}");
```